### ISSUE TITLE
Batch queries with different `include` when the `where` is the same

### ISSUE THREAD

**Issue by andresribeiro** [prisma/prisma#16374]
## Problem

Prisma could batch queries with different `include` when the `where` is the same.

For example, two queries occurring on the same tick, one only selecting `name` of the user with id 1, while the other is only selecting `username` of the user with id 1. Currently, 2 different queries are sent to the database, but it could be just one, selecting `name` and `username` of the user with id 1 at once.

This can allow an optimization of GraphQL resolvers, where each resolver takes only the field that is of interest to it, and Prisma optimizes all these fields in a single query, not selecting from the database any field that has not been requested by the client

## Suggested solution

Batch these queries

**Comment by ArthurLobopro** [comment]
+1

**Comment by janpio** [comment]
Doesn't that have the danger that the additional `include` values could e.g. blow up the size of _many_ queries if only 1 of them requests a lot more than all the others?

**Comment by andresribeiro** [comment]
> Doesn't that have the danger that the additional `include` values could e.g. blow up the size of _many_ queries if only 1 of them requests a lot more than all the others?

Definitely. I know that's a hard thing to do and maybe it's not even worth, just an idea which could be very useful for people building GraphQL APIs. If you think it's not worth, I'd complete understand and close this issue
